components/widgets/inputs/users_invite_input.jsx

```diff
@@ -0,0 +1,331 @@
+import {isGuest} from 'utils/utils';
+
+import './users_emails_input.scss';
+
+export default class UsersEmailsInput extends React.PureComponent {
+    static propTypes = {
+        placeholder: PropTypes.string,
+        ariaLabel: PropTypes.string.isRequired,
+        usersLoader: PropTypes.func,
+        onChange: PropTypes.func,
+        showError: PropTypes.bool,
+        errorMessageId: PropTypes.string,
+        errorMessageDefault: PropTypes.string,
+        errorMessageValues: PropTypes.object,
+        value: PropTypes.arrayOf(PropTypes.oneOfType([PropTypes.object, PropTypes.string])),
+        onInputChange: PropTypes.func,
+        inputValue: PropTypes.string,
+        noMatchMessageId: PropTypes.string,
+        noMatchMessageDefault: PropTypes.string,
+        validAddressMessageId: PropTypes.string,
+        validAddressMessageDefault: PropTypes.string,
+        loadingMessageId: PropTypes.string,
+        loadingMessageDefault: PropTypes.string,
+        emailInvitationsEnabled: PropTypes.bool,
+    }
+
+    static defaultProps = {
+        noMatchMessageId: t('widgets.users_emails_input.no_user_found_matching'),
+        noMatchMessageDefault: 'No one found matching **{text}**, type email address',
+        validAddressMessageId: t('widgets.users_emails_input.valid_email'),
+        validAddressMessageDefault: 'Add **{email}**',
+        loadingMessageId: t('widgets.users_emails_input.loading'),
+        loadingMessageDefault: 'Loading',
+        showError: false,
+    };
+
+    constructor(props) {
+        super(props);
+        this.selectRef = React.createRef();
+        this.state = {
+            options: [],
+        };
+    }
+
+    renderUserName = (user) => {
+        const parts = getLongDisplayNameParts(user);
+        let fullName = null;
+        if (parts.fullName) {
+            fullName = (<span className='fullname'>{parts.fullName}</span>);
+        }
+        let nickname = null;
+        if (parts.nickname) {
+            nickname = (<span className='nickname'>{parts.nickname}</span>);
+        }
+
+        return (
+            <>
+                {parts.displayName}
+                {fullName}
+                {nickname}
+            </>
+        );
+    }
+
+    loadingMessage = () => {
+        const text = (
+            <FormattedMessage
+                id={this.props.loadingMessageId}
+                defaultMessage={this.props.loadingMessageDefault}
+            />
+        );
+
+        return (<LoadingSpinner text={text}/>);
+    }
+
+    getOptionValue = (user) => {
+        return user.id || user.value;
+    }
+
+    formatOptionLabel = (user, options) => {
+        const profileImg = imageURLForUser(user.id, user.last_picture_update);
+        let guestBadge = null;
+        let botBadge = null;
+
+        if (user.is_bot) {
+            botBadge = <BotBadge/>;
+        }
+
+        if (!isEmail(user.value) && isGuest(user)) {
+            guestBadge = <GuestBadge/>;
```

Comment:
can we do this inline? I know this is an existent practice in mattermost but we have discussed this to be only for complex component changes.